src/main/java/io/cdap/plugin/dlp/SensitiveRecordFilter.java

```diff
@@ -0,0 +1,310 @@
+    Schema inputSchema = stageConfigurer.getInputSchema();
+
+    try {
+
+      client = DlpServiceClient.create(getSettings());
+    } catch (IOException e) {
+      LOG.error(e.printStackTrace());
+    }
+
+    config.validate(stageConfigurer.getFailureCollector(), inputSchema);
+
+    Map<String, Schema> outputs = new HashMap<>();
+    outputs.put(SENSITIVE_PORT, inputSchema);
+    outputs.put(NON_SENSITIVE_PORT, inputSchema);
+    stageConfigurer.setOutputSchemas(outputs);
+  }
+
+  /**
+   * Initialize this <code>SensitiveRecordFilter</code> plugin. A instance of DLP client is created with mapped
+   * infotypes.
+   *
+   * @param context Initialization context
+   */
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+
+    client = DlpServiceClient.create(getSettings());
+    metrics = context.getMetrics();
+  }
+
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+
+    String templateName = String.format("projects/%s/inspectTemplates/%s", config.getProject(), config.templateId);
+    GetInspectTemplateRequest request = GetInspectTemplateRequest.newBuilder().setName(templateName).build();
+
+    try {
+      InspectTemplate template = client.getInspectTemplate(request);
+    } catch (Exception e) {
+      throw new IllegalArgumentException(
+        "Unable to validate template name. Ensure template ID matches the specified ID in DLP");
+    }
+  }
+
+  /**
+   * Splitter Transform splits the sensitive and non-sensitive record into different ports. If user has selected entire
+   * record to be checked for sensitive data, then all the fields are concacted as string and passed  to data loss
+   * prevention API.
+   *
+   * @param record  a <code>StructuredRecord</code> being passed from the previous stage.
+   * @param emitter a <code>MultiOutputEmitter</code> to emit sensitive or non-sensitive data on different ports.
+   */
+  @Override
+  public void transform(StructuredRecord record, MultiOutputEmitter<StructuredRecord> emitter) throws Exception {
+    String recordString = null;
+    ContentItem contentItem = null;
+
+    if (!config.entireRecord) {
+      recordString = record.get(config.getFieldName()).toString();
+    } else {
+      recordString = StructuredRecordStringConverter.toDelimitedString(record, ",");
+    }
+    try {
+
+      // depending on input schema field object
+      contentItem = ContentItem.newBuilder().setValue(recordString).build();
+
+      String templateName = String.format("projects/%s/inspectTemplates/%s", config.getProject(), config.templateId);
+
+      InspectContentRequest request =
+        InspectContentRequest.newBuilder()
+          .setParent(ProjectName.of(config.getProject()).toString())
+          .setInspectTemplateName(templateName)
+          .setItem(contentItem)
+          .build();
+
+      metrics.count("dlp.requests.count", 1);
```

Comment:
can you add the de scription on what does this metric mean in the comment and why is it needed? Also add it in the plugin doc?

Reply:
done